beets/util/__init__.py

```diff
@@ -500,6 +500,32 @@ def link(path, dest, replace=False):
                               traceback.format_exc())
 
 
+def hardlink(path, dest, replace=False):
+    """Create a hard link from path to `dest`. Raises an OSError if
+    `dest` already exists, unless `replace` is True. Does nothing if
+    `path` == `dest`."""
+    if (samefile(path, dest)):
+        return
+
+    path = syspath(path)
+    dest = syspath(dest)
+    if os.path.exists(dest) and not replace:
+        raise FilesystemError(u'file exists', 'rename', (path, dest))
+    try:
+        os.link(path, dest)
+    except NotImplementedError:
+        # raised on python >= 3.2 and Windows versions before Vista
+        raise FilesystemError(u'OS does not support hard links.'
+                              'link', (path, dest), traceback.format_exc())
+    except OSError as exc:
+        # TODO: Windows version checks can be removed for python 3
+        if hasattr('sys', 'getwindowsversion'):
+            if sys.getwindowsversion()[0] < 6:  # is before Vista
+                exc = u'OS does not support hard links.'
+        raise FilesystemError(exc, 'link', (path, dest),
+                              traceback.format_exc())
+
+
```

Comment:
this part about windows version less than 6 needs to go. I don't think they are supported on windows at all, on any version.